types/node/test.d.ts

```diff
@@ -442,5 +455,190 @@ declare module 'node:test' {
         timeout?: number | undefined;
     }
 
-    export { test as default, run, test, describe, it, before, after, beforeEach, afterEach };
+    interface MockCommonOptions {
+        /**
+         * The number of times that the mock will use the behavior of `implementation`.
+         * Once the mock function has been called `times` times,
+         * it will automatically restore the behavior of `original`.
+         * This value must be an integer greater than zero.
+         * @default Infinity
+         */
+        times?: number | undefined;
+    }
+
+    interface MockFunctionOptions extends MockCommonOptions {}
+
+    interface MockMethodOptions extends MockCommonOptions {
+        /**
+         * If `true`, `object[methodName]` is treated as a getter.
+         * This option cannot be used with the `setter` option.
+         */
+        getter?: boolean | undefined;
+
+        /**
+         * If `true`, `object[methodName]` is treated as a setter.
+         * This option cannot be used with the `getter` option.
+         */
+        setter?: boolean | undefined;
+    }
+
+    type Mock<F extends Function> = F & {
+        mock: MockFunctionContext<F>;
+    };
+
+    type NoOpFunction = (...args: any[]) => undefined;
+
+    type FunctionPropertyNames<T> = {
+        [K in keyof T]: T[K] extends Function ? K : never;
+    }[keyof T];
+
+    interface MockTracker {
+        /**
+         * This function is used to create a mock function.
+         * @param original An optional function to create a mock on.
+         * @param implementation An optional function used as the mock implementation for `original`.
+         *  This is useful for creating mocks that exhibit one behavior for a specified number of calls and then restore the behavior of `original`.
+         * @param options Optional configuration options for the mock function.
+         */
+        fn<F extends Function = NoOpFunction>(original?: F, options?: MockFunctionOptions): Mock<F>;
+        fn<F extends Function = NoOpFunction>(original?: F, implementation?: F, options?: MockFunctionOptions): Mock<F>;
+
+        /**
+         * This function is used to create a mock on an existing object method.
+         * @param object The object whose method is being mocked.
+         * @param methodName The identifier of the method on `object` to mock. If `object[methodName]` is not a function, an error is thrown.
+         * @param implementation An optional function used as the mock implementation for `object[methodName]`.
+         * @param options Optional configuration options for the mock method.
+         */
+        method<
+            MockedObject extends object,
+            MethodName extends FunctionPropertyNames<MockedObject>,
```

Comment:
👀 `FunctionPropertyNames` and `Mock` now don't work for getters and setters.

https://github.com/nodejs/node/blob/d375242f8b0d4f36c6eb940423995d853b08b7ce/test/parallel/test-runner-mocking.js#L244-L301

WIP